pkg/controller/servicecidrs/servicecidrs_controller.go

```diff
@@ -332,7 +340,14 @@ func (c *Controller) sync(ctx context.Context, key string) error {
 		return err
 	}
 
-	// Set Ready condition to True.
+	// Set Ready condition to True if necessary
+	for _, condition := range cidr.Status.Conditions {
+		if condition.Type == networkingapiv1.ServiceCIDRConditionReady {
+			if condition.Status == metav1.ConditionTrue {
+				return nil
```

Comment:
I dislike the random `return nil`s... it's not immediately obvious that "return" means "skip the status update, but don't skip anything else".

This function might benefit from splitting out an "update the ServiceCIDRConditionReady condition if it needs to be updated" function. (Especially if I was wrong about the existence of a helper function.)

Reply:
ack